--- a/daemon.c
+++ b/daemon.c
@@ -113,7 +113,7 @@ void logmsg(int prio, const char *fmt, .
  */
 struct in_addr get_remote(struct svc_req *rqstp)
 {
-    return (svc_getcaller(rqstp->rq_xprt))->sin_addr;
+    return ((struct sockaddr_in*)(svc_getcaller(rqstp->rq_xprt)))->sin_addr;
 }
 
 /*
@@ -121,7 +121,7 @@ struct in_addr get_remote(struct svc_req
  */
 short get_port(struct svc_req *rqstp)
 {
-    return (svc_getcaller(rqstp->rq_xprt))->sin_port;
+    return ((struct sockaddr_in*)(svc_getcaller(rqstp->rq_xprt)))->sin_port;
 }
 
 /*
